src/java/org/apache/cassandra/tools/nodetool/Status.java

```diff
@@ -170,24 +215,134 @@ private void addNode(String endpoint, Float owns, HostStatWithPort hostStat, int
         strOwns = owns != null && hasEffectiveOwns ? new DecimalFormat("##0.0%").format(owns) : "?";
         hostID = hostIDMap.get(endpoint);
 
-        try
-        {
+        try {
             rack = epSnitchInfo.getRack(endpoint);
-        }
-        catch (UnknownHostException e)
-        {
+        } catch (UnknownHostException e) {
             throw new RuntimeException(e);
         }
 
-        epDns = hostStat.ipOrDns(printPort);
         if (isTokenPerNode)
-        {
-            tableBuilder.add(statusAndState, epDns, load, strOwns, hostID, hostStat.token, rack);
-        }
+            return List.of(addressAndPort, statusAndState, epDns, load, strOwns, hostID, hostStat.token, rack);
         else
-        {
-            tableBuilder.add(statusAndState, epDns, load, String.valueOf(size), strOwns, hostID, rack);
+            return List.of(addressAndPort, statusAndState, epDns, load, String.valueOf(size), strOwns, hostID, rack);
+    }
+
+    // To check for descending order
+    private Boolean desc() {
+        return sortOrder == null ? null : sortOrder == SortOrder.desc;
+    }
+
+    // Sort function to sort the data
+    private Map<String, List<Object>> sort(Map<String, List<Object>> map) {
+        switch (sortBy) {
+            case none:
+                return map;
+            case ip:
+                return sortByIp(map, 0, desc() != null ? desc() : false);
```

Comment:
you can simplify this down to `return sortByIp(map)` because 

1) you know what index to look for in the particular method you call, you do not need to specify it here, because there is never a case where you call some sorting method twice, each time with different index
2) you can call `desc()` in the sorting method itself, again, you do not need to pass it here.

Reply:
I will update the changes accordingly as you suggested